cpp/src/gandiva/selection_vector.h

```diff
@@ -34,22 +34,22 @@ class SelectionVector {
   virtual ~SelectionVector() = default;
 
   /// Get the value at a given index.
-  virtual uint GetIndex(int index) const = 0;
+  virtual uint64_t GetIndex(int64_t index) const = 0;
```

Comment:
Should these all be signed int64?

Reply:
The values inside the selection vector are all unsigned - that way, a 16-bit selection vector can work with batch size upto 64K-1 (instead of 32K - 1).

but, if are anyway casting to a 64-bit type, I guess it doesn't matter.